src/bitmaps.rs

```diff
@@ -0,0 +1,199 @@
+use std::borrow::Borrow;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::Path;
+use std::{fs, io};
+
+use endio_bit::BitReader;
+
+use allsorts::binary::read::ReadScope;
+use allsorts::bitmap::{self, BitDepth, BitmapSize, CBDTTable, CBLCTable, GlyphBitmapData};
+use allsorts::fontfile::FontFile;
+use allsorts::tables::FontTableProvider;
+use allsorts::tag::{self};
+
+use crate::cli::BitmapOpts;
+use crate::BoxError;
+
+pub fn main(opts: BitmapOpts) -> Result<i32, BoxError> {
+    let buffer = std::fs::read(&opts.font)?;
+    let scope = ReadScope::new(&buffer);
+    let font_file = scope.read::<FontFile>()?;
+    let table_provider = font_file.table_provider(opts.index)?;
+
+    let table = table_provider
+        .table_data(tag::CBLC)?
+        .or(table_provider.table_data(tag::EBLC)?)
+        .ok_or("font does not have CBLC or EBLC tables")?;
+    let scope = ReadScope::new(table.borrow());
+    let cblc = scope.read::<CBLCTable<'_>>().unwrap();
+    let table = table_provider
+        .table_data(tag::CBDT)?
+        .or(table_provider.table_data(tag::EBDT)?)
+        .ok_or("font does not have CBDT or EBDT tables")?;
+    let scope = ReadScope::new(table.borrow());
```

Comment:
Why does this live have to be done twice?

Reply:
Different tables EB**LC** vs EB**DT**

Reply:
Ah. Cool, I didn't see that.